src/controller.py

```diff
@@ -1,6 +1,6 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
-
+# pylint: disable=too-many-lines
```

Comment:
This is probably a sign that we need to split in different module each controller

Reply:
I agree, but I think it is out of scope for this PR, maybe opening another PR with refactoring will do, what do you think?

Reply:
Well this PR is leading to this pylint warning, so that seems to be in the scope of that PR. I don't think this will require a big refactoring, but just moving each controller in a dedicated module. Does it sound ok ?

Reply:
Ok, I'll refactor in this PR then :+1: